doc/man1/openssl-cmp.pod.in

```diff
@@ -0,0 +1,1142 @@
+
+=item B<-srv_untrusted>
+
+Intermediate certs for constructing chains for CMP protection by client.
+
+=item B<-rsp_cert>
+
+Certificate to be returned as mock enrollment result.
+
+=item B<-rsp_extracerts>
+
+Extra certificates to be included in mock certification responses.
+
+=item B<-rsp_capubs>
+
+CA certificates to be included in mock ip response.
+
+=item B<-poll_count>
+
+Number oftimes the client must poll before receiving a certificate.
+
+=item B<-check_after>
+
+The checkAfter value (time to wait) to include in poll response.
+
+
+=item B<-grant_implicitconf>
+
+Grant implicit confirmation of newly enrolled certificate.
+
+=item B<-pkistatus>
+
+PKIStatus to be included in server response.
+
+=item B<-failure>
+
+A single failure info bit number to be included in server response.
+
+=item B<-failurebits>
+
+Number representing failure bits to be included in server response.
+
+=item B<-statusstring>
+
+Status string to be included in server response.
+
+=item B<-send_error>
+
+Force server to reply with error message.
+
+=item B<-send_unprotected>
+
+Send response messages without CMP-level protection.
+
+=item B<-send_unprot_err>
+
+In case of negative responses, server shall send unprotected error messages,
+certificate responses (ip/cp/kup), and revocation responses (rp).
+WARNING: This setting leads to behavior violating RFC 4210.
+
+=item B<-accept_unprotected>
+
+Accept missing or invalid protection of requests.
+
+=item B<-accept_unprot_err>
+
+Accept unprotected error messages from client.
+
+=item B<-accept_raverified>
+
+Accept RAVERIFED as proof-of-possession (POPO).
+
+=back
+
+
+=head1 NOTES
+
+When setting up CMP configurations and experimenting with enrollment options
+typically various errors occur until the configuration is correct and complete.
+When the CMP server reports an error the client will by default
+check the protection of the CMP response message.
+Yet some CMP services, in particular those based on EJBCA,
+tend not to protect negative responses.
+In this case the client will reject them, and thus their contents are not shown
+although they usually contain hints that would be helpful for diagnostics.
+For assisting in such cases the CMP client offers a workaround via the
+B<-unprotected_errors> option, which allows accepting such negative messages.
+
+
+=head1 EXAMPLES
+
+=head2 Simple examples using the default OpenSSL configuration file
+
+This CMP client implementation comes with demonstrative CMP sections
+in the example configuration file F<openssl/apps/openssl.cnf>,
+which can be used to interact conveniently with the Insta Demo CA.
+
+In order to enroll an initial certificate from that CA it is sufficient
+to issue the following shell commands.
+
+  cd openssl
+  export OPENSSL_CONF=apps/openssl.cnf
+  apps/openssl cmp -section insta
+
+This should produce the file F<insta.cert.pem> containing a new certificate
+for the private key held in F<test/insta.priv.pem>.
```

Comment:
Again a reference to the source tree. Will we be installing this key somewhere?

Reply:
I don't think it's worth the effort installing the key and the Insta root CA cert.
I've added instead commands to generate such a key and to download that cert.